experimenter/run_pipeline.py

```diff
@@ -137,19 +137,38 @@ def run(self, pipeline: Pipeline, metric_names: list = None) -> list:
             metrics = get_metrics_from_problem_description(problem_description)
         else:
             metrics = get_metrics_from_list(metric_names)
-
-        all_scores, all_results = evaluate(
-            pipeline,
-            inputs,
-            data_pipeline=data_pipeline,
-            scoring_pipeline=scoring_pipeline,
-            problem_description=problem_description,
-            data_params=data_params,
-            metrics=metrics,
-            context=context,
-            volumes_dir=self.volumes_dir,
-            runtime_environment=runtime_environment,
-        )
+        if (self.random_seed is None):
+            all_scores, all_results = evaluate(
+                pipeline,
+                inputs,
+                data_pipeline=data_pipeline,
+                scoring_pipeline=scoring_pipeline,
+                problem_description=problem_description,
+                data_params=data_params,
+                metrics=metrics,
+                context=context,
+                volumes_dir=self.volumes_dir,
+                runtime_environment=runtime_environment,
+            )
+        #for seed change
+        else:
+            logger.warning('Random Seed: {}'.format(self.random_seed))
+            all_scores, all_results = evaluate(
+                pipeline,
+                inputs,
+                data_pipeline=data_pipeline,
+                scoring_pipeline=scoring_pipeline,
+                problem_description=problem_description,
+                data_params=data_params,
+                metrics=metrics,
+                context=context,
+                volumes_dir=self.volumes_dir,
+                runtime_environment=runtime_environment,
+                random_seed=self.random_seed,
+                scoring_random_seed=self.random_seed, 
+                data_random_seed=self.random_seed
+            )
+        
```

Comment:
this is a lot of duplicated code. lets set variables based on conditions and then have only one call to `evaluate`.